### ISSUE TITLE
FileNotFound Error on starting local webhost 

### ISSUE THREAD

**Issue by Rik-de-Kort** [Azure/azure-functions-python-worker#108]
Following instructions from [here](https://github.com/Azure/azure-functions-python-worker/wiki/Setting-up-development-environment). In particular, I'm interested in having a local version of my Azure function available for development and testing. I'd like to use Python 3.x for this (why Python 2.7 is the default I'll never understand).
I'm quite new to Azure, so not always sure where to look for what yet.

Platform I'm running on is Windows 10, Python distribution is Python 3.6.4, ran from an Anaconda prompt in an isolated conda environment (called azure).

The first issue arises when I execute ```python setup.py webhost```. Downloads fine, but then it can't find a file.
````
(azure) C:\Users\User\Documents\lab\project_repo\azure-functions-python-worker>python setup.py webhost
running webhost
Downloading Azure Functions Web Host...
Extracting Azure Functions Web Host binaries...
error: [WinError 2] The system cannot find the file specified
````
There are still some files in the /build/worker/ folder, so I decided to try and run ````python -m azure.worker.testutils```` However, that leads to an error. After manually downloading the functions binaries with the provided link, I copy-pasted the entire contents of the folder in /build/worker/ (seemed like the right thing to do), but still the error remained.

````
(azure) C:\Users\User\Documents\lab\project_repo\azure-functions-python-worker>python -m azure.worker.testutils C:\Users\User\Documents\lab\project_repo\
Traceback (most recent call last):
  File "C:\Users\User\Anaconda3\envs\azure\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\User\Anaconda3\envs\azure\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\User\Documents\lab\project_repo\azure-functions-python-worker\azure\worker\testutils.py", line 563, in <module>
    _main()
  File "C:\Users\User\Documents\lab\project_repo\azure-functions-python-worker\azure\worker\testutils.py", line 555, in _main
    script_root=os.path.abspath(args.scriptroot))
  File "C:\Users\User\Documents\lab\project_repo\azure-functions-python-worker\azure\worker\testutils.py", line 509, in popen_webhost
    stderr=stderr)
  File "C:\Users\User\Anaconda3\envs\azure\lib\subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "C:\Users\User\Anaconda3\envs\azure\lib\subprocess.py", line 997, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] The system cannot find the file specified
````

I went around and hacked in `print(dll)` right before the `popen_webhost`, but it shows the correct path to the dll, which exists. So either it's not finding a file which does exist, or it's looking for something else that doesn't exist. I'm not sure which one.

Any ideas? Am I even using an up-to-date method? Should I use something different entirely (there's the `func` command on PowerShell which I also have installed)?

**Comment by elprans** [comment]
@Rik-de-Kort Do you have [azure-functions-core-tools](https://github.com/Azure/azure-functions-core-tools) v2 installed?

**Comment by Rik-de-Kort** [comment]
Yes. Exact versions:
Azure Functions Core Tools (220.0.0-beta.0)
Function Runtime Version: 2.0.11651.0

Sorry for the delay in response, long weekend due to King's Day here.

**Comment by asavaritayal** [comment]
@Rik-de-Kort please try the updated instructions here - https://github.com/Azure/azure-functions-python-worker/wiki/Create-Function-(CLI)

**Comment by asavaritayal** [comment]
@Rik-de-Kort can you confirm if you're unblocked here, so we can close the issue?

**Comment by Rik-de-Kort** [comment]
I've been away on vacation. Will have a go today and let you know. :)

Thanks for the help.

**Comment by Rik-de-Kort** [comment]
1. I installed 2.0.1-beta26 via `npm i azure-functions-core-tools@core`. `npm ls` confirms this gives me 2.0.1-beta26
2. When creating a new function project via `func init`, it says I'm limited to C# or JavaScript as langs.
3. Trying to hack together an `__init__.py` with `function.json` as per [the programming model](https://github.com/Azure/azure-functions-python-worker/blob/dev/docs/index.rst) yields `[2018-05-22 1:09:49 PM] testfunc: Object reference not set to an instance of an object.` after running `func host start`. (Full output below).
4. I cannot find anything whatsoever indicating the `--worker-runtime python` argument did anything. It sure isn't part of the local directory.

Since I'm still having issues, but they seem to be of a different kind, would you prefer to close this and have a new issue, or keep this open?

<details>
<summary>Full output</summary>
````
PS C:\Users\User\documents\datalab\test> func host start

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Hosting environment: Production
Content root path: C:\Users\User\documents\datalab\test
Now listening on: http://localhost:7071
Application started. Press Ctrl+C to shut down.
[2018-05-22 1:09:48 PM] Reading host configuration file 'C:\Users\User\documents\datalab\test\host.json'
[2018-05-22 1:09:48 PM] Host configuration file read:
[2018-05-22 1:09:48 PM] {}
[2018-05-22 1:09:48 PM] Starting Host (HostId=desktopljhdee2-1618153932, InstanceId=f7220dbc-6ff9-4ee0-9a0c-cd1460342f04, Version=2.0.11651.0, ProcessId=54056, AppDomainId=1, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
[2018-05-22 1:09:49 PM] Unable to configure java worker. Could not find JAVA_HOME app setting.
[2018-05-22 1:09:49 PM]
[2018-05-22 1:09:49 PM] Could not configure language worker Java.
[2018-05-22 1:09:49 PM]
[2018-05-22 1:09:49 PM] Generating 0 job function(s)
[2018-05-22 1:09:49 PM] No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. config.UseServiceBus(), config.UseTimers(), etc.).
[2018-05-22 1:09:49 PM] Host initialized (532ms)
[2018-05-22 1:09:49 PM] Host started (571ms)
[2018-05-22 1:09:49 PM] Job host started
[2018-05-22 1:09:49 PM] The following 1 functions are in error:
[2018-05-22 1:09:49 PM] testfunc: Object reference not set to an instance of an object.
[2018-05-22 1:09:49 PM]
[2018-05-22 1:09:49 PM]
Listening on http://localhost:7071/
Hit CTRL-C to exit...
````
</details>

**Comment by asavaritayal** [comment]
> 2. When creating a new function project via func init, it says I'm limited to C# or JavaScript as langs.

@Rik-de-Kort for #2, since Python is still under private preview, you will need to specify `--worker-runtime python` with `func init`. Please refer to the instructions here - https://github.com/Azure/azure-functions-python-worker/wiki/Create-Function-(CLI)

> 3. Trying to hack together an __init__.py with function.json as per the programming model yields [2018-05-22 1:09:49 PM] testfunc: Object reference not set to an instance of an object. after running func host start. (Full output below).

`func new` should automatically populate the __init__.py and `function.json` for you. 

> 4. I cannot find anything whatsoever indicating the --worker-runtime python argument did anything. It sure isn't part of the local directory.

It is recommended that you use the most recent version of the CLI (beta28). `npm update azure-functions-core-tools@core`

> Since I'm still having issues, but they seem to be of a different kind, would you prefer to close this and have a new issue, or keep this open?

I'm closing the issue for now but please feel free to open a new one if you still run into any problems.
